### PR TITLE
ARROW-15601: [Docs][Release] Update post release script to move stable docs + keep dev docs

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
This updates the post-release script for the docs with:

- keep the `/docs/dev` docs
- move the current stable docs (without all versioned + dev docs subdirectories) to a new versioned subdirectory

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
One thing I just realized is that the "move current stable to previous version" logic only needs to happen for a major release, and not when doing a bug-fix release (eg 7.0.1). So that logic should maybe be behind a flag (eg based on whether the second "previous_version" parameter was specified?)

**Comment by kou** [comment]
How about checking the minor and patch version in the first argument?

```shell
case "${version}" in
  *.0.0)
    is_major_release=yes
    ;;
  *)
    is_major_release=no
    ;;
esac
```

If we need to change the number of arguments depending on context, a human error will occur.

**Comment by jorisvandenbossche** [comment]
@kou thanks for the suggestion!

File: dev/release/post-09-docs.sh [hunk 1]
```diff
@@ -43,22 +53,39 @@ git checkout -b asf-site origin/asf-site
...
+# list and remove previous versioned docs
 versioned_paths=()
 for versioned_path in docs/*.0/; do
   versioned_paths+=(${versioned_path})
+  rm -rf ${versioned_path}
 done
+# add to list and remove dev docs
+versioned_paths+=("docs/dev/")
+rm -rf docs/dev/
+if [ "$is_major_release" = true ] ; then
```
Comment (kou) [inline dev/release/post-09-docs.sh:65]:
```suggestion
if [ "$is_major_release" = "yes" ] ; then
```

File: dev/release/post-09-docs.sh [hunk 2]
```diff
@@ -43,22 +53,39 @@ git checkout -b asf-site origin/asf-site
...
 curl \
   --fail \
   --location \
   --remote-name \
   https://apache.jfrog.io/artifactory/arrow/docs/${version}/docs.tar.gz
+cp ../docs.tar.gz .
 tar xvf docs.tar.gz
 rm -f docs.tar.gz
 git checkout docs/c_glib/index.html
+if [ "$is_major_release" = true ] ; then
```
Comment (kou) [inline dev/release/post-09-docs.sh:83]:
```suggestion
if [ "$is_major_release" = "yes" ] ; then
```

File: dev/release/post-09-docs.sh [hunk 3]
```diff
@@ -43,22 +53,39 @@ git checkout -b asf-site origin/asf-site
...
+fi
+# delete current stable docs and restore all previous versioned docs
 rm -rf docs/*
 git checkout "${versioned_paths[@]}"
 curl \
   --fail \
   --location \
   --remote-name \
   https://apache.jfrog.io/artifactory/arrow/docs/${version}/docs.tar.gz
+cp ../docs.tar.gz .
```
Comment (kou) [inline dev/release/post-09-docs.sh:79]: Is this needed?
It seems that `../docs.tar.gz` doesn't exist.

File: dev/release/post-09-docs.sh [hunk 4]
```diff
@@ -43,22 +53,39 @@ git checkout -b asf-site origin/asf-site
...
 tar xvf docs.tar.gz
 rm -f docs.tar.gz
 git checkout docs/c_glib/index.html
+if [ "$is_major_release" = true ] ; then
+  mv docs_temp docs/${previous_version}
+fi
 git add docs
 git commit -m "[Website] Update documentations for ${version}"
+git clean -d -f -x
```
Comment (kou) [inline dev/release/post-09-docs.sh:88]: Is this needed?

File: dev/release/post-09-docs.sh (hunk 3 above)
Comment (jorisvandenbossche) [reply dev/release/post-09-docs.sh:79]: Whoops, that was from local testing (to not download the archive on every test run), will remove.

File: dev/release/post-09-docs.sh (hunk 4 above)
Comment (jorisvandenbossche) [reply dev/release/post-09-docs.sh:88]: Not strictly "needed" for the creating the correct PR, but the script does leave a bunch of ignored files after running it locally (eg we don't commit the .doctrees files), and including this ensures it cleans up after itself. I could also limit it to the /docs/ directory (or remove it altogether)

Comment (kou) [reply dev/release/post-09-docs.sh:88]: Ah, I see.

**Comment by jorisvandenbossche** [comment]
Sorry for the delay here, this should be ready now I think

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**